sample/src/main/java/com/google/modernstorage/sample/mediastore/MediaStoreViewModel.kt

```diff
@@ -21,129 +21,113 @@ import androidx.lifecycle.AndroidViewModel
+        val request = Request.Builder().url(url).build()
 
-            withContext(Dispatchers.IO) {
-                val response = httpClient.newCall(request).execute()
+        return withContext(Dispatchers.IO) {
+            val response = httpClient.newCall(request).execute()
 
-                response.body?.use { responseBody ->
-                    val filename = generateFilename(MediaSource.INTERNET, "mp4")
-                    val videoUri = mediaStore.addVideoFromStream(
-                        filename = filename,
-                        inputStream = responseBody.byteStream(),
-                        location = SharedPrimary
-                    )
+            val mediaUri = response.body?.use { responseBody ->
+                val filename = generateFilename(MediaSource.INTERNET, extension)
 
-                    withContext(Dispatchers.Main) {
-                        callback(videoUri)
-                    }
+                return@use mediaStore.addMediaFromStream(
+                    filename = filename,
+                    type = type,
+                    mimeType = mimeType,
+                    inputStream = responseBody.byteStream(),
+                    location = SharedPrimary
+                ).getOrElse {
+                    return@withContext Result.failure(it)
                 }
-            }
+            } ?: return@withContext Result.failure(Exception("Could not download this media"))
+
+            mediaStore.getResourceByUri(mediaUri)
+                .apply {
+                    savedStateHandle.set(CURRENT_MEDIA_KEY, this.getOrNull())
+                    setLoadingStatus(false)
+                }
+                .onSuccess { return@withContext Result.success(it) }
+                .onFailure { return@withContext Result.failure(it) }
         }
     }
 
-    fun createMediaUriForCamera(type: MediaType, callback: (uri: Uri) -> Unit) {
-        viewModelScope.launch {
-            val uri = when (type) {
-                MediaType.IMAGE -> mediaStore.createImageUri(
-                    generateFilename(MediaSource.CAMERA, "jpg"),
-                    SharedPrimary
-                )
-                MediaType.VIDEO -> mediaStore.createVideoUri(
-                    generateFilename(MediaSource.CAMERA, "mp4"),
-                    SharedPrimary
-                )
-            }
-
-            withContext(Dispatchers.Main) {
-                if (uri != null) callback(uri)
-            }
+    suspend fun createMediaUriForCamera(type: MediaType): Result<Uri> {
+        return when (type) {
+            MediaType.IMAGE -> mediaStore.createMediaUri(
+                filename = generateFilename(MediaSource.CAMERA, "jpg"),
+                type = FileType.IMAGE,
+                location = SharedPrimary
+            )
+            MediaType.VIDEO -> mediaStore.createMediaUri(
+                generateFilename(MediaSource.CAMERA, "mp4"),
+                type = FileType.IMAGE,
```

Comment:
```suggestion
                type = FileType.VIDEO,
```
FileType should be `VIDEO`

Reply:
Copy pasting isn't always a good thing. Thanks for the review 😅

Reply:
@lukasvalkovic Could you sign the Google CLA as mentioned here? https://github.com/google/modernstorage/pull/24#issuecomment-866345745